crates/holochain_cascade/src/lib.rs

```diff
@@ -82,11 +82,19 @@ macro_rules! some_or_return {
     };
 }
 
+/// Marks whether data came from a local store or another node on the network
+pub enum CascadeSource {
+    /// Data came from a local store
+    Local,
+    /// Data came from another node on the network
+    Network,
+}
```

Comment:
I can't think of any reason why this would be relevant, except in testing scenarios, and it may even be misleading... If we retrieved something through gossip a microsecond ago, does that really count as "Local"?

Reply:
Or imagine an optimization where the cache won't go out to the network if a query has been made over the network in the last X timeframe... That would make this return value relatively useless, no?

Reply:
@neonphog I only added this because [there was already a code path in production](https://github.com/holochain/holochain/blob/d1153885d838439018240b7b5be96720887bb21d/crates/holochain/src/core/sys_validate.rs#L705-L728) that was simulating this by initializing a local-only cascade, trying to get something from it, then initializing a full cascade with the network and trying to get something with that, purely so that it could be determined whether it was "gotten from the network" or not, and [taking different actions accordingly](https://github.com/holochain/holochain/blob/d1153885d838439018240b7b5be96720887bb21d/crates/holochain/src/core/sys_validate.rs#L499-L507).

So maybe that logic just needs to be updated. This is the only place it's used, if that's rewritten then I can remove this.

Reply:
I agree with your assessment though. It would be nice if Holochain were built around pubsub instead of fetch, so that you're only ever getting things locally, and you can at times refresh your local cache so that when you go to get things locally, what you want to be there will have a chance of being there. That would solve the misleadingness.